pkg/web/wslisten.go

```diff
@@ -43,6 +43,20 @@ func (w *WrappedSocketListener) Addr() net.Addr {
 	return w.addr
 }
 
+func (w *WrappedSocketListener) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	if atomic.LoadInt32(&w.closed) != 0 {
+		http.Error(rw, http.StatusText(http.StatusServiceUnavailable), http.StatusServiceUnavailable)
+		return
+	}
+	w.logger.Printf("Upgrading web connection from: %v", r.RemoteAddr)
```

Comment:
An alternative to keeping a pointer to the logger might be to get it like this:
```go
r.Context().Value(http.ServerContextKey).(*http.Server).ErrorLog
```
(and handle if it's nil). I don't know which of these you prefer.